app/main/views/suppliers.py

```diff
@@ -33,6 +33,14 @@ def get_supplier(code):
 
     supplier = DataAPIClient().get_supplier(code)['supplier']
 
+    try:
+        if supplier['recruiter'] == 'no' and len(supplier['domains']['assessed']) == 0:
+            if ((current_user.role != 'supplier' and current_user.supplier_code != code) and
+               current_user.role not in ['admin', 'assessor']):
```

Comment:
I think this can be simplified if you merge all the conditions into the first `if` statement.

Reply:
Yes, it can be simplified. I have pushed a change to rectify this.